File: ruby/grains/grains.rb
class Grains
  def self.square(n)
    squares = [1]

    (0..62).each {|num|
      double = squares[num]*2
      squares << double
    }
    @squares_array = squares
    squares[n-1]
  end

  def self.total
    @squares_array.reduce(:+)
  end

end

File: javascript/grains/grains.js
var BigInt = require('./big-integer');

var Grains = function() {}

Grains.prototype.square = function(n) {
  return BigInt(2).pow(n-1).toString();
}

//sum equals 2^n - 1
Grains.prototype.total = function() {
  return BigInt(2).pow(64).prev().toString();
}

module.exports = Grains;